File: src/main/java/com/example/demo/controller/MainController.java
package com.example.demo.controller;

import com.example.demo.util.TimesheetGenerator;
import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.time.YearMonth;
import java.util.List;
import java.util.regex.Pattern;

@Controller
@Validated
public class MainController {

    private static final Pattern YM_PATTERN = Pattern.compile("\\d{4}-\\d{2}");

    @GetMapping("/")
    public String index() {
        return "redirect:/tools";
    }

    @GetMapping("/tools")
    public String tools(Model model) {
        model.addAttribute("screens", List.of());
        model.addAttribute("currentScreen", "ツール");
        model.addAttribute("selectedScreenName", "ツール");
        model.addAttribute("currentScreenPath", "toolsList");
        return "main";
    }

    @GetMapping("/timesheet")
    public String timesheet(
            @RequestParam(name = "month", required = false) String monthParam,
            Model model,
            Authentication principal
    ) {
        YearMonth yearMonth = YearMonth.now();
        if (monthParam != null && YM_PATTERN.matcher(monthParam).matches()) {
            try {
                yearMonth = YearMonth.parse(monthParam);
            } catch (Exception ignored) {
            }
        }
        var dates = TimesheetGenerator.generateDates(yearMonth);
        model.addAttribute("currentScreen", "勤務表");
        model.addAttribute("selectedScreenName", "勤務表");
        model.addAttribute("currentScreenPath", "timesheetMonth");
        model.addAttribute("monthDisplay", TimesheetGenerator.formatYearMonth(yearMonth));
        model.addAttribute("yearMonthValue", yearMonth.toString());
        model.addAttribute("dates", dates);
        model.addAttribute("currentUserName", principal.getName());
        return "main";
    }

    @GetMapping("/tools/password")
    public String passwordTool(Model model) {
        model.addAttribute("screens", List.of());
        model.addAttribute("currentScreen", "パスワード生成");
        model.addAttribute("selectedScreenName", "パスワード生成");
        model.addAttribute("currentScreenPath", "passwordGeneration");
        return "main";
    }

    @GetMapping("/tools/passkey")
    public String passkeyTool(Model model, Authentication principal) {
        model.addAttribute("screens", List.of());
        model.addAttribute("currentScreen", "パスキー");
        model.addAttribute("selectedScreenName", "パスキー");
        model.addAttribute("currentScreenPath", "passkeyTools");
        if (principal != null) {
            model.addAttribute("currentUserName", principal.getName());
        }
        return "main";
    }
}

File: src/main/kotlin/com/example/demo/MainController.kt
package com.example.demo

import com.example.demo.constants.ApplicationConstants
import com.example.demo.util.TimesheetGenerator
import org.springframework.security.core.Authentication
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import java.time.YearMonth

@Controller
@Validated
class MainController {
    @GetMapping(ApplicationConstants.ROOT)
    @Suppress("SpringMVCViewInspection")
    fun index(): String = "${ApplicationConstants.REDIRECT}/tools"

    @GetMapping("/tools")
    fun tools(model: Model): String {
        model.addAttribute("screens", emptyList<Any>())
        model.addAttribute("currentScreen", "ツール")
        model.addAttribute("selectedScreenName", "ツール")
        model.addAttribute("currentScreenPath", "toolsList")
        return "main"
    }

    @GetMapping("/timesheet")
    fun timesheet(
        @RequestParam(name = "month", required = false) monthParam: String?,
        model: Model,
        principal: Authentication
    ): String {
        val yearMonth = monthParam
            ?.takeIf { it.matches(Regex("\\d{4}-\\d{2}")) }
            ?.let {
                runCatching { YearMonth.parse(it) }.getOrElse { YearMonth.now() }
            } ?: YearMonth.now()
        val dates = TimesheetGenerator.generateDates(yearMonth)
        model.apply {
            addAttribute("currentScreen", "勤務表")
            addAttribute("selectedScreenName", "勤務表")
            addAttribute("currentScreenPath", "timesheetMonth")
            addAttribute("monthDisplay", TimesheetGenerator.formatYearMonth(yearMonth))
            addAttribute("yearMonthValue", yearMonth.toString())
            addAttribute("dates", dates)
            addAttribute("currentUserName", principal.name)
        }
        return "main"
    }

    @GetMapping("/tools/password")
    fun passwordTool(model: Model): String {
        model.addAttribute("screens", emptyList<Any>())
        model.addAttribute("currentScreen", "パスワード生成")
        model.addAttribute("selectedScreenName", "パスワード生成")
        model.addAttribute("currentScreenPath", "passwordGeneration")
        return "main"
    }

    @GetMapping("/tools/passkey")
    fun passkeyTool(model: Model, principal: Authentication?): String {
        model.addAttribute("screens", emptyList<Any>())
        model.addAttribute("currentScreen", "パスキー")
        model.addAttribute("selectedScreenName", "パスキー")
        model.addAttribute("currentScreenPath", "passkeyTools")
        principal?.name?.let { model.addAttribute("currentUserName", it) }
        return "main"
    }
}
